src/components/content/link.tsx

```diff
@@ -6,35 +6,74 @@ import { StyledA } from '../tags/styled-a'
 import { ExternalLink } from './external-link'
 
 export interface LinkProps {
-  element: {
-    href: string
-  }
+  href?: string
   children: React.ReactNode
+  className?: string
+  noExternalIcon?: boolean
+  title?: string
 }
 
-export function Link({ element, children = null }: LinkProps) {
+//TODO: Should come from cloudflare worker
+const legacyLinks = [
+  '/entity/unrevised',
+  '/auth/login',
+  '/auth/logout',
+  '/privacy',
+  '/imprint',
+  '/terms',
+  '/disable-frontend',
+  '/enable-frontend',
+]
+
+export function Link({
+  href,
+  children,
+  className,
+  noExternalIcon,
+  title,
+}: LinkProps) {
   const prettyLinks = React.useContext(PrettyLinksContext)
 
-  if (!element.href) return <>{children}</>
+  if (!href || href === undefined || href === '') return <>{children}</>
+
+  const prettyLink = getPrettyLink(href)
+  const displayHref = prettyLink ? prettyLink : href
+
+  const isAbsolute = href.indexOf('//') > -1
+  const isExternal = isAbsolute && href.indexOf('.serlo.org') === -1
+  const isLegacyLink = legacyLinks.indexOf(displayHref) > -1
+
+  if (isExternal || (isAbsolute && prettyLink === undefined))
+    return renderLink()
+  if (!isLegacyLink || prettyLink) return renderClientSide()
 
-  const isExternal = element.href.indexOf('//') > -1
-  const prettyLink =
-    prettyLinks !== undefined
-      ? prettyLinks[element.href.replace('/', 'uuid')]?.alias
-      : undefined
+  //fallback
+  return renderLink()
+
+  function getPrettyLink(href: string): string | undefined {
+    if (prettyLinks === undefined || prettyLinks === {}) return undefined
+
+    const prettyLink = prettyLinks[href.replace('/', 'uuid')]?.alias
+    if (prettyLink !== undefined) return prettyLink
+
+    //fallback for wrong absolute links
+    return prettyLinks['uuid' + href.split('de.serlo.org/')[1]]?.alias
```

Comment:
I'd probably move that together with the Context into its own hook `const prettyifyLink = usePrettyLink()` (or something). This way, you have the logic in one place, we could unit test it and possibly include it somewhere else (though a single Link component is probably enough for the last point).

Reply:
Also, the whole `isAbsolute` handling. Would be pretty nice if we have a hook that accepts a whatever url and tells you if it can be rendered client side and which href to use.

Reply:
I guess I get your point (although I never really worked with unit tests before), but isn't this what this refactor did?
The only place where PrettyLinksContext is consumed and checked currently is this component (same for the `isAbsolute` stuff).

Reply:
If you still think it's useful I'll go and read about custom hooks :)